mParticle-Adjust/MPKitAdjust.m

```diff
@@ -132,5 +132,13 @@ - (void)adjustAttributionChanged:(nullable ADJAttribution *)attribution {
     [_kitApi onAttributionCompleteWithResult:attributionResult error:nil];
 }
 
+- (MPKitExecStatus *)setATTStatus:(MPATTAuthorizationStatus)status withATTStatusTimestampMillis:(NSNumber *)attStatusTimestampMillis  API_AVAILABLE(ios(14)){
+    [Adjust requestTrackingAuthorizationWithCompletionHandler:^(NSUInteger status) {
```

Comment:
It's important that our SDK doesn't end up forcing the user to show the prompt. By doing this, it will trigger the prompt whenever the customer sets the status, even if the status is just set to not_determined on SDK startup.

I think a better mapping would be to just set the status on adjust - it looks like they have a property we could set? 
https://github.com/adjust/ios_sdk/blob/master/Adjust/Adjust.h#L292

Reply:
Hey guys, Ugi here from Adjust. 👋

@samdozor Is right, calling that wrapper method will cause ATT dialog to be shown. That method is added to Adjust SDK as a convenience method for clients and in addition to that, upon providing an answer to ATT dialog shown via Adjust wrapper method, we will communicate new ATT status right away to our backend.

Noticed that your method is called set*. There is no way to set ATT status in Adjust SDK. Method from line 292 is also just a convenient getter we added for people to query for current state of ATT status. Adjust SDK is trying to read and determine value of ATT status on its own, no way to inject it.

Also noticed that @BrandonStalnaker is mentioning v4.23 which doesn't contain this ATT getter, that's added later.

Cheers

Reply:
@samdozor That's a read only property. They don't provide any other interface for us to se the value. I could change it so that we don't perform this if the value is 'not_determined'